src/node.c

```diff
@@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
   if (node == NULL || child == NULL) {
     return false;
   }
-  if (NODE_MEM(node) != NODE_MEM(child)) {
```

Comment:
I just pushed the fixed version, which calls:

```
res = cmark_node_new_with_mem(CMARK_NODE_TEXT, parser->mem);
```

however as I explained this will not be an option for externally-defined extensions.